crates/sui-types/src/messages.rs

```diff
@@ -825,7 +825,7 @@ impl Message for SenderSignedData {
     type DigestType = TransactionDigest;
 
     fn digest(&self) -> Self::DigestType {
-        TransactionDigest::new(sha3_hash(self))
+        TransactionDigest::new(sha3_hash(&self.data))
     }
 
     fn verify(&self) -> SuiResult {
```

Comment:
Now that we aren't pinning user sigs this way, we may want to get rid of the `SenderSignedData` type and replace it with an envelope wrapper.

Reply:
good point, it can come after https://github.com/MystenLabs/sui/pull/6445 that updates the SenderSignedData with intent message.